core-java/src/main/java/com/baeldung/ZipCollectionExample.java

```diff
@@ -0,0 +1,41 @@
+package com.baeldung.guava.tutorial;
+
+import com.google.common.collect.Streams;
+import org.jooq.lambda.Seq;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.IntStream;
+
+public class ZipCollectionExample {
+    static List<String> names = new ArrayList<>(Arrays.asList("John", "Jane", "Jack", "Dennis"));
+
+    static List<Integer> ages = new ArrayList<>(Arrays.asList(24, 25, 27));
+
+    public static void main(String[] args) {
+        //      Using Streams API from Guava 21
+        Streams
```

Comment:
Separate samples should be organized in the form of short test cases. When it's done like this, readers can run all of them separately easily